DependencyInjection/HttplugExtension.php

```diff
@@ -359,94 +356,34 @@ private function createUri(ContainerBuilder $container, $serviceId, $uri)
     private function configureAutoDiscoveryClients(ContainerBuilder $container, array $config)
     {
         $httpClient = $config['discovery']['client'];
+        if ($httpClient !== 'auto') {
+            $container->removeDefinition('httplug.auto_discovery.auto_discovered_client');
+            $container->removeDefinition('httplug.collector.auto_discovered_client');
+            $container->removeDefinition('httplug.auto_discovery.auto_discovered_client.plugin');
+        }
 
-        if (!empty($httpClient)) {
-            if ($httpClient === 'auto') {
-                $httpClient = $this->registerAutoDiscoverableClient(
-                    $container,
-                    'auto_discovered_client',
-                    $this->configureAutoDiscoveryFactory(
-                        $container,
-                        HttpClientDiscovery::class,
-                        'auto_discovered_client',
-                        $config
-                    ),
-                    $this->isConfigEnabled($container, $config['profiling'])
-                );
-            }
-
-            $httpClient = new Reference($httpClient);
+        if (!empty($httpClient) && $httpClient !== 'auto') {
```

Comment:
I don't used to write nested conditions (object calisthenics :heart:) but I will make an exception.